src/TreeBuilder.cpp

```diff
@@ -626,15 +616,15 @@ void TreeBuilder::processContainer(const Variable& variable, Node& node) {
       node.pointer = next();
       containerStats.length = *node.pointer ? 1 : 0;
       containerStats.capacity = 1;
-      if (!shouldProcess(*node.pointer)) {
+      if (bool followPointer = next() == 1; !followPointer) {
```

Comment:
I have a hard time _following_ this line... The negation makes it confusing to understand at a glance.
Using some constants could help in this case. What about something like:
```cpp

/* Tag indicating if the pointer has  been followed or skipped */
enum class TrackPointerTag: uintptr_t {
  /* The content has been skipped.
   * It prevents double counting the footprint of a node the JIT code has seen before,
   * double counting content being stored inline, or getting stuck in an infinite loop
   * when processing circular linked list.
   */
  skipped = 0,
  followed = 1,
};

if (next() == TrackPointerTag::skipped) return;
```